renku_notebooks/api/classes/server.py

```diff
@@ -221,40 +222,21 @@ def _get_registry_secret(self, b64encode=True):
         return output
 
     def _get_session_k8s_resources(self):
-        cpu_request = float(self.server_options["cpu_request"])
-        mem = self.server_options["mem_request"]
-        gpu_req = self.server_options.get("gpu_request", {})
-        gpu = {"nvidia.com/gpu": str(gpu_req)} if gpu_req else None
+        cpu_request = float(self.server_options.cpu)
+        mem = self.server_options.memory
+        gpu_req = self.server_options.gpu
+        gpu = {"nvidia.com/gpu": str(gpu_req)} if gpu_req > 0 else None
         resources = {
             "requests": {"memory": mem, "cpu": cpu_request},
             "limits": {"memory": mem},
         }
         if config.sessions.enforce_cpu_limits == "lax":
-            if "cpu_request" in config.server_options.ui_choices:
-                resources["limits"]["cpu"] = max(
-                    config.server_options.ui_choices["cpu_request"]["options"]
-                )
-            else:
-                resources["limits"]["cpu"] = cpu_request
+            resources["limits"]["cpu"] = 3 * cpu_request
```

Comment:
What's the `3` here?

Reply:
This is to allow the cpu to spike up momentarily. Because if we set the limit on the cpu it gets throttled and then when you create a new session it takes a really long time for the session to start up. If the limits are set to "lax" (i.e. "relaxed") then the limit is larger than the request. But if the limits are set to strict then the limit is the same as the request and the cpu is throttled.

Reply:
makes sense. you should give it a descriptive name, random magic numbers in code aren't nice :slightly_smiling_face: